src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -547,6 +547,23 @@ void BossAI::UpdateAI(uint32 diff)
     DoMeleeAttackIfReady();
 }
 
+void BossAI::_DespawnAtEvade()
+{
+    uint32 corpseDelay = me->GetCorpseDelay();
+    uint32 respawnDelay = me->GetRespawnDelay();
+
+    me->SetCorpseDelay(1);
```

Comment:
Wouldn't be necessary if you use `me->SetRespawnTime(seconds)` after `me->DespawnOrUnsummon()` though, which IMO is the proper way (instead of using a hack and temporarily changing respawn delay).